tests.py

```diff
@@ -181,6 +228,39 @@ def test_cli_generic_args(args, attribute, expected):
     assert getattr(anonip.parse_arguments(args), attribute) == expected
 
 
+@pytest.mark.parametrize(
+    "args,success",
+    [
+        ([], True),
+        (["--regex", "test"], True),
+        (["-c", "4"], True),
+        (["--regex", "test", "-c", "3"], False),
+        (["--regex", "test", "-l", ";"], False),
+        (["--regex", "test", "-l", ";", "-c", "4"], False),
+    ],
+)
+def test_cli_args_ambiguity(args, success):
+    if success:
+        anonip.parse_arguments(args)
+        return
+
+    with pytest.raises(SystemExit) as e:
+        anonip.parse_arguments(args)
+    assert e.value.code == 2
+
+
+@pytest.mark.parametrize(
+    "args,expected",
+    [
+        (["--regex", "test"], "test"),
+        (["--regex", "foo", "bar", "baz"], "foo|bar|baz"),
```

Comment:
does this still work as intended if e.g. the first provided regex is foo|ter? i think then the concatenated regex is foo|ter|bar|baz, which is maybe not what the user expected. maybe we should group each provided regex before concaneation?

Reply:
Could you please provide an example where this actually is an issue? I can imagine that just joining them with `|` is too lazy, but I fail to see the problem.

Reply:
I am not aware of a good example now. It just popped up while reading the code :). Maybe this simple approach its good enough for now.